lib/go-atscfg/parentdotconfig.go

```diff
@@ -383,258 +390,112 @@ func makeParentDotConfigData(
-
-					pasvc.Parents = []*ParentAbstractionServiceParent{&ParentAbstractionServiceParent{
-						FQDN:   pasvc.DestDomain,
-						Port:   pasvc.Port,
-						Weight: 1.0,
-					}}
+				topo.Nodes = append(topo.Nodes, node)
 
-					// text += `dest_domain=` + orgURI.Hostname() + ` port=` + orgURI.Port() + ` go_direct=true` + "\n"
-				} else {
-
-					// check for profile psel.qstring_handling.  If this parameter is assigned to the server profile,
-					// then edges will use the qstring handling value specified in the parameter for all profiles.
-
-					// If there is no defined parameter in the profile, then check the delivery service profile.
-					// If psel.qstring_handling exists in the DS profile, then we use that value for the specified DS only.
-					// This is used only if not overridden by a server profile qstring handling parameter.
+				for _, cg := range cgnames {
+					topo.Nodes = append(topo.Nodes, tc.TopologyNode{Cachegroup: cg})
+				}
+			}
 
-					// TODO refactor this logic, hard to understand (transliterated from Perl)
-					dsQSH := queryStringHandling
-					if dsQSH == nil {
-						dsQSH = ParentSelectParamQStringHandlingToBool(dsParams.QueryStringHandling)
-						if dsQSH == nil && dsParams.QueryStringHandling != "" {
-							warnings = append(warnings, "Delivery Service parameter '"+ParentConfigParamQStringHandling+"' value '"+dsParams.QueryStringHandling+"' malformed, not using!")
-						}
+			nameTopologies[TopologyName(topoName)] = topo
+			ds.Topology = util.StrPtr(topoName)
+		}
 
-					}
-					parentQStr := dsQSH
-					if parentQStr == nil {
-						v := false
-						parentQStr = &v
-					}
-					if ds.QStringIgnore != nil && tc.QStringIgnore(*ds.QStringIgnore) == tc.QStringIgnoreUseInCacheKeyAndPassUp && dsQSH == nil {
-						v := true
-						parentQStr = &v
-					}
-					if parentQStr == nil {
-						b := !DefaultIgnoreQueryStringInParentSelection
-						parentQStr = &b
-					}
+		isMSO := ds.MultiSiteOrigin != nil && *ds.MultiSiteOrigin
 
-					pasvc.DestDomain = orgURI.Hostname()
-					pasvc.Port, err = strconv.Atoi(orgURI.Port())
-					if err != nil {
-						if strings.ToLower(orgURI.Scheme) == "https" {
-							pasvc.Port = 443
-						} else {
-							pasvc.Port = 80
-						}
-						warnings = append(warnings, "DS '"+*ds.XMLID+"' had malformed origin  port: '"+orgURI.Port()+"': using "+strconv.Itoa(pasvc.Port)+"! : "+err.Error())
-					}
-					pasvc.Parents = parents
-					pasvc.SecondaryParents = secondaryParents
-					pasvc.SecondaryMode = secondaryMode
-					pasvc.RetryPolicy = roundRobin
-					pasvc.GoDirect = goDirect
-					pasvc.IgnoreQueryStringInParentSelection = !*parentQStr
-					// text += `dest_domain=` + orgURI.Hostname() + ` port=` + orgURI.Port() + ` ` + parents + ` ` + secondaryParents + ` ` + roundRobin + ` ` + goDirect + ` qstring=` + parentQStr + "\n"
-				}
+		// TODO put these in separate functions. No if-statement should be this long.
+		if ds.Topology == nil || *ds.Topology == "" {
```

Comment:
With the fake topology you're adding above, is it ever possible for this to be true? Should this block be removed?

Reply:
Removed as it does seem like checks would call continue before this code would be hit.